### PR TITLE
Add TimeseriesDB and integrate it to chained-function-eventing

### PR DESCRIPTION
**Pull request by boramalper**
**Known Bugs:**
- Duration is always zero.
- Crash in `timeseriesdb` if an experiment has not been started while events arrive.

See [Knative Eventing μBenchmarks Documentation](https://github.com/boramalper/vhive/blob/utils-benchmarking-eventing/function-images/tests/chained-function-eventing/README.md) for a demo.

Signed-off-by: Bora M. Alper <bora@boramalper.org>

### CODE REVIEW

**Comment by boramalper** [comment]
@ustiugov There are some minor bugs to iron out, but the design decisions are here to stay and I would appreciate your feedback on the overall architecture of TimeseriesDB, taking our conversation on Slack in consideration too (which I will also add to its documentation).

File: utils/benchmarking/eventing/relay/main.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+func UnmarshalVHiveMetadata(s string) *proto.VHiveMetadata {
+	var j struct {
+		Id        string `json:"Id"`
+		InvokedOn string `json:"InvokedOn"`
+	}
+	if err := json.Unmarshal([]byte(s), &j); err != nil {
+		log.Fatalf("failed to unmarshal vhivemetadata: %s", err)
+	}
+	invokedOnT, err := time.Parse("2006-01-02T15:04:05-0700", j.InvokedOn)
```
Comment (ustiugov) [inline utils/benchmarking/eventing/relay/main.go:48]: why this constant?

**Review by ustiugov — COMMENTED** [review]
- [x] use regular services instead of Knative
- [x] time precision should be in microseconds. use this format: https://github.com/ease-lab/vhive/blob/67b50287ef17f477c40e85fec2d99ed319e5eb88/vhive.go#L102
- [ ] invoker needs to be modified as well

File: function-images/tests/chained-function-eventing/README.md [hunk 2]
```diff
@@ -43,6 +51,39 @@ Manifests are split into separate files and are enumerated in the order of their
...
+   tscli timeseriesdb.default.192.168.1.240.sslip.io:80 end xxx
+   ```
+
+   **Sample Output:**
+   ```
+   INVOCATION {b8a155f8-343a-4c0b-a1bc-f087d9c4a89b}
+   ==================================================
+   Id       : b8a155f8-343a-4c0b-a1bc-f087d9c4a89b
+   InvokedOn: 2021-06-17T19:41:56+0000
+   Duration : 0s
```
[the remark this answers is not in the document]
Comment (boramalper) [reply function-images/tests/chained-function-eventing/README.md:65]: Zero value is formatted as `0s` whereas smaller values are formatted correctly automatically (e.g. `5ms` or `234ns`) depending on the value; see: https://golang.org/pkg/time/#Duration.String

If you want a fixed format (e.g. `XXXms`) at all times, let me know.

File: utils/benchmarking/eventing/relay/main.go (hunk 1 above)
Comment (boramalper) [reply utils/benchmarking/eventing/relay/main.go:48]: Go uses a specific date to format timestamps (where other languages use printf-style formatters such as `%Y-%m-%dT%H:%M:%S`). This is [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) standard, but I will switch to [`time.RFC3339Nano`](https://golang.org/pkg/time/#pkg-constants) instead for better precision.

File: function-images/tests/chained-function-eventing/manifests/2-ksvc.yaml [hunk 3]
```diff
@@ -1,3 +1,49 @@
+# Knative Service: timeseriesdb
+# =============================
+apiVersion: serving.knative.dev/v1
```
[the remark this answers is not in the document]
Comment (boramalper) [reply function-images/tests/chained-function-eventing/manifests/2-ksvc.yaml:3]: Do regular K8s services have a fixed IP address that I can hard-code in my programs? For instance with Knative, I know that timeseriesdb will always be at `timeseriesdb.default.192.168.1.240.sslip.io:80`.

[the remark this answers is not in the document]
Comment (ustiugov) [reply function-images/tests/chained-function-eventing/manifests/2-ksvc.yaml:3]: yes, here is an example of such manifest
https://github.com/ease-lab/vhive/blob/main/configs/storage/minio/service.yaml

File: function-images/tests/chained-function-eventing/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (ustiugov) [reply function-images/tests/chained-function-eventing/README.md:65]: fixed yes

**Comment by boramalper** [comment]
Closing in favour of #286 for CI/CD tests.

**Pull request closed without merging**